### DISCUSSION TITLE
有谁会写这个代理配置文件吗？

### DISCUSSION THREAD

**Discussion by onehuhuhu** [GaiZhenbiao/ChuanhuChatGPT#419] [Q&A]
我找了半天不会配这个代理配置文件，有没有大佬给指点一下如何写

**Comment by SongAoFei** [comment]
如果你有机场的话，直接在机场配置文件里面直接添加就好了（在Profiles页面找到机场配置，右键点Edit），然后记得不要开系统代理，只开TUN，记得Service Mode安装一下

**Reply by onehuhuhu** [reply]
我一直开的系统代理，一直报错我的IP是中国，我问chatgpt改了改配置文件，但是还是没弄好，不知道是不是一直开这个系统代理的原因

**Reply by SongAoFei** [reply]
关了系统代理，只开TUN你试试

**Reply by GaiZhenbiao** [reply]
检测IP在服务器启动时进行，而不是网页刷新时。

**Reply by onehuhuhu** [reply]
rule-providers:
private:
type: http
behavior: domain
url: "https://cdn.jsdelivr.net/gh/Loyalsoldier/clash-rules@release/lancidr.txt"
path: ./ruleset/ads.yaml
interval: 86400
我将这些内容放在机场配置文件的最上面

RULE-SET,private,DIRECT
DOMAIN-SUFFIX,openai.com,<你的代理规则>
将这两行放在rules:这个里面的最上面，<你的代理规则>这部分我改成了机场配置文件proxy-groups:里面的一个name，这样写对吗？

**Reply by SongAoFei** [reply]
<你的代理规则>这部分我是仿照的其他的写的，我看其他的域名代理规则都是机场的名字，我就也这样写
其实好像不写也可以，我的直接使用机场的那个配置也可以
你先自己运行试试
不行就慢慢调

**Reply by onehuhuhu** [reply]
谢谢，我再研究研究

**Reply by caishu07** [reply]
老哥研究怎么样了，我也一直error，甚至直接提示目标服务器拒绝，怎么配置clash都没用，求救啊，也不会在clash配置里添加那些代码，到底在哪添加啊？？

**Reply by onehuhuhu** [reply]
> 老哥研究怎么样了，我也一直error，甚至直接提示目标服务器拒绝，怎么配置clash都没用，求救啊，也不会在clash配置里添加那些代码，到底在哪添加啊？？

rule-providers:
private:
type: http
behavior: domain
url: "https://cdn.jsdelivr.net/gh/Loyalsoldier/clash-rules@release/lancidr.txt"
path: ./ruleset/ads.yaml
interval: 86400
我将这些内容放在机场配置文件的最上面
在机场的rules里面的最上面添加
IP-CIDR,127.0.0.1/32,DIRECT
RULE-SET,private,DIRECT
DOMAIN-SUFFIX,openai.com,GW云洞
<你的代理规则>这部分我改成了我的机场proxy-groups:里面的一个name，也就是gw云洞，看官方文档说是我的127.0.0.1没有直连所以加了一条直连的配置。我这样配置完，就可以运行了。我也是仅开tun模式的。刚开始仅开tun模式无法上网，不清楚为什么，过了一会可以联网了，就能正常运行了。现在openai封号严重，记得用稳定的机场，不要选择亚洲节点，选择美国节点比较好。

**Reply by caishu07** [reply]
> > 老哥研究怎么样了，我也一直error，甚至直接提示目标服务器拒绝，怎么配置clash都没用，求救啊，也不会在clash配置里添加那些代码，到底在哪添加啊？？
> 
> rule-providers: private: type: http behavior: domain url: "https://cdn.jsdelivr.net/gh/Loyalsoldier/clash-rules@release/lancidr.txt" path: ./ruleset/ads.yaml interval: 86400 我将这些内容放在机场配置文件的最上面 在机场的rules里面的最上面添加 IP-CIDR,127.0.0.1/32,DIRECT RULE-SET,private,DIRECT DOMAIN-SUFFIX,openai.com,GW云洞 <你的代理规则>这部分我改成了我的机场proxy-groups:里面的一个name，也就是gw云洞，看官方文档说是我的127.0.0.1没有直连所以加了一条直连的配置。我这样配置完，就可以运行了。我也是仅开tun模式的。刚开始仅开tun模式无法上网，不清楚为什么，过了一会可以联网了，就能正常运行了。现在openai封号严重，记得用稳定的机场，不要选择亚洲节点，选择美国节点比较好。

我目前情况是删掉congfig打开，手动输入apikey才能正常使用，这说明我的网络配置应该没问题，折腾一上午clash的设置脑子都炸了实在不知道错在哪，因为一开始我按照你的方式把这些代码输入机场配置，保存后直接提示错误不能打开，不得已就删了另想他法，最后只能得到目前这个手动输入key使用的方式

**Reply by onehuhuhu** [reply]
和配置确实没有关系，我试了试，添加好配置，和不添加配置，都能正常运行

**Reply by caishu07** [reply]
又在主力机上试了试，还是一样的结果，我换v2ray开tun模式，仔细检查了虚拟网卡没啥问题似乎，照样直接把api输入进去也链接不上必须先打开网页后输入ip，估计还是哪里没有被tun照顾到流量没走代理（你的qq搜索显示查无此人(灬ꈍ ꈍ灬)）

**Reply by onehuhuhu** [reply]
在文件中设定默认密钥、用户名密码以及更多设置（推荐）
这样设置的密钥以及其他设置项可以在拉取项目更新之后保留。

在项目文件夹中复制一份 config_example.json，并将其重命名为 config.json，在其中填入 API-Key、用户名密码（可选）、API host（可选）、代理地址（可选）等设置。用户名密码支持多用户。示例：

{
	"openai_api_key": "sk-xxxxxxxxxxxxxxxxxxxxxxxxx",
	"users": [
		["用户1的用户名", "用户1的密码"],
		["用户2的用户名", "用户2的密码"]
	], 
}
如果不设置用户名与密码，可以直接将“users”字段整段删去，或留空为 "users": [], 
这里的配置密钥你配置好了吗？

**Reply by onehuhuhu** [reply]
我qq隐私设置了，你再加一下，

**Comment by learnCodeON** [comment]
不需要写那个代理配置文件，我开启TUN模式就能使用。

**Reply by onehuhuhu** [reply]
我将那个配置文件添加到机场配置文件里面了，不知道是配置文件起来作用还是tun起来作用，我也能正常运行了，看来应该是由于这个tun起作用了。不过后来登入ChatGPT的时候，给我报红了，吓我一跳我以为我的号被封了，后来也能正常登入。

**Reply by caishu07** [reply]
我开了也没用，求救啊

**Reply by learnCodeON** [reply]
> 我开了也没用，求救啊

如果你想本地部署这个项目，clash需要用收费的机场。如果你不想花这个钱也行，可以用clash的免费的节点（github上面有，请搜索clash免费节点），在别人部署好的网站中使用。具体方法是：在Profiles中把你找到的免费节点下载并且使用，在Proxies中开启直连（有的地方叫direct），在General中打开TUN Mode。最后，你可以访问huggingface.co/spaces/JohnSmith9982/ChuanhuChatGPT，这是别人部署好的项目（你也可以自己找），你只需要把 api key 输入就能用了。如果你使用我这个方法，你的ip还是在国内。还有，不要问我上面的英文是什么意思，你自己去找翻译软件。

**Reply by caishu07** [reply]
> > 我开了也没用，求救啊
> 
> 如果你想本地部署这个项目，clash需要用收费的机场。如果你不想花这个钱也行，可以用clash的免费的节点（github上面有，请搜索clash免费节点），在别人部署好的网站中使用。具体方法是：在Profiles中把你找到的免费节点下载并且使用，在Proxies中开启直连（有的地方叫direct），在General中打开TUN Mode。最后，你可以访问huggingface.co/spaces/JohnSmith9982/ChuanhuChatGPT，这是别人部署好的项目（你也可以自己找），你只需要把 api key 输入就能用了。如果你使用我这个方法，你的ip还是在国内。还有，不要问我上面的英文是什么意思，你自己去找翻译软件。

我买了的很贵的kycloud，clash也打开了tun，开了全局模式，打开还是提示目标计算机拒绝，后来我去blibili看，有人遭遇我一样的问题，说是把config里的https和http删掉其中一个就能开，我删掉了http确实打开了，但是输入文字后它还是提示error，没法连接，彻底抓瞎了；后来我又按照上边的题主的方法把那些代码直接黏贴进机场配置文件里，结果保存提示错误，我确定没黏贴错，一时间不知道该怎么办了；最后，我直接把congfig删了，打开网页，手动输入apikey这才能正常用，所以我觉得不是我的梯子配置问题，是congfig里有问题，不知道大佬们遭遇过这种情况没有？虽然手动输入api也能用，但我是布置在家里捡的一台e5洋垃圾上，希望能穿透出去用手机或者在办公室访问，所以还是想要不需要手动输入api的访问方式，目前仍旧抓瞎中……

**Reply by GaiZhenbiao** [reply]
试着更新到最新版代码。我自己在用config中设置的方式，没有问题。注意，应该设置`openai_api_key`字段。

**Reply by caishu07** [reply]
> 试着更新到最新版代码。我自己在用config中设置的方式，没有问题。注意，应该设置`openai_api_key`字段。

上午刚下的最新版，没用，还是得删了config，打开网页后手动输入key才能正常用，不知道错在哪

**Comment by yanyuxiyangzk** [comment]
我的服务部署在国内，希望添加自定义代理去访问openapi接口，这个自定义代理是怎么个实现方法呢，自定义代理地址如何填写和配置呢

**Reply by onehuhuhu** [reply]
常见问题里面有配置文件的写法，我是将给的写法加入机场的配置文件中，再将你的代理规则，改为机场的名字，

**Reply by onehuhuhu** [reply]
但是我不确定是我的代理文件写对了，还是开启tun模式的原因，最终我可以运行了

**Comment by yanyuxiyangzk** [comment]
我国内服务的机器上不希望安装clash，可以实现吗

**Reply by onehuhuhu** [reply]
https://github.com/noobnooc/noobnooc/discussions/9  你看看这篇文章是否能为你提供帮助

**Comment by yanyuxiyangzk** [comment]
开启tun模式，就直接可以访问到openapi接口了，这样做不是很安全，我把服务搭建在国内，可以做小程序用。openapi通过配置代理方式访问openapi接口，让用户自己去填写代理地址，除了cloudflare做api代理外，还有什么更好的方法呢，当然服务直接搭建在海外，省事省力，但小程序就需要调用接口的话，就需要额外开发接口。

**Reply by onehuhuhu** [reply]
抱歉，我不太懂

**Comment by yanyuxiyangzk** [comment]
我按照你发的方法，已经通过了，3中方式可以选择。谢谢啊
